consensus/consensus_v2.go

```diff
@@ -521,6 +521,10 @@ func (consensus *Consensus) commitBlock(blk *types.Block, committedMsg *FBFTMess
 	atomic.AddUint64(&consensus.blockNum, 1)
 	consensus.SetCurBlockViewID(committedMsg.ViewID + 1)
 	consensus.LeaderPubKey = committedMsg.SenderPubkeys[0]
```

Comment:
I think the better solution is to move 521-523 above OnConsensusdone(), rather than move that code from OnConsensusDone to here.

Reply:
nvm. this works too. approved. These code needs a bit more restructure anyway in the pipelining code.

Reply:
It's possible that `OnConsensusDone` returns an error. If we move `OnConsensusDone` (which calls `InsertChain`),  the node might be blocked in consensus module since the blockNum has been incremented while the block is not added in blockchain. 

Also, `UpdateConsensusInformation` would better not to be exposed to public. Please let me know your further thinking.

Reply:
Yea, please let me know if there are any restructure ideas. I would be super happy to be engaged in the discussion.